projects/extension/src/background/ConnectionManager.ts

```diff
@@ -105,20 +105,32 @@ export class ConnectionManager extends (EventEmitter as { new(): StateEmitter })
    * by a content script.
    */
   addApp(port: chrome.runtime.Port): void {
+    let notifMsg: string;
     if (!this.#client) {
       throw new Error('Smoldot client does not exist.');
     }
     const app = this.#apps.find(s => s.name === port.name);
     if (app) {
+      notifMsg = `App ${port.name} already exists.`;
       port.postMessage({ type: 'error', payload: `App ${port.name} already exists.` });
     } else {
       const app = new AppMediator(port, this as ConnectionManagerInterface)
       // if associate fails by returning false it has sent an error down the
       // port and disconnected it, so we should just discard this `AppMediator`
       if (app.associate()) {
         this.registerApp(app);
+        const appInfo = port.name.split('::');
+        notifMsg = `App ${appInfo[0]} connected to ${appInfo[1]}.`
       }
     }
+    chrome.storage.sync.get('notifications', (s) => {
+      s.notifications && chrome.notifications.create('', {
```

Comment:
Can you add a test in the `ConnectionManager` tests that mocks the chrome storage API and spies on `chrome.notifications.create` to verify this behaviour happens please?

Reply:
Reading [this](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/notifications/create)  I think you should probably be using the name of the port as the `notificationId` .. this would clear any previous notifications about that port before displaying this one.

Reply:
I think changes to the `ConnectionManager` should be accompanied by tests.  There's already test infrastructure in place to test it.

Reply:
> Im not sure why I need the notification Id. (also Im using chrome API) In the example of Multi-network connect there are 3 notifications appearing one after the after denoting that the app connected to each different relay chain (polkadot, kusama, westend).

I don't think you *need* it but it doesn't feel right having an empty string. Using the port name as the id was my first instinct because it identifies the channel uniquely and it might prove useful debugging later.  You'll still get 3 notifications if you use the port name because the ids (for multi-network-demo) will be: `Multiple Network Demo::polkadot`, `Multiple Network Demo::kusama` and `Multiple Network Demo::westend`.

The MDN I linked to gets mapped to the chrome API, it's the same thing as `chrome.notifications.create`.  I just tend to prefer using the MDN documentation because:

1. Firefox is the lowest common denominator, it has the smallest API support out of chrome / firefox
2. It's easy to accidentally end up on the manifest v3 docs in chrome, which we don't support because of Firefox.

Reply:
> I think changes to the ConnectionManager should be accompanied by tests. There's already test infrastructure in place to test it.

Yup - I deleted that part of my comment when I actually realised your point - but Github issues was polite enough to leave it there :) I will proceed with the test as it is irrelevant to the actual notification (that I want to test on #225 ) and more relevant to the connection manager action